### PR TITLE
VideoBackends: add support to allow rendering to multiple output targets

### PR DESCRIPTION
**Pull request by iwubcode**
This was used in #10362 and will be used in future features.  Splitting it out to cut down on scope and make it easier to review

### CODE REVIEW

**Comment by iwubcode** [comment]
Marked as a draft while I look into a fifoci bug

File: Source/Core/VideoCommon/GXPipelineTypes.h [hunk 1]
```diff
@@ -19,7 +19,7 @@ namespace VideoCommon
 // As pipelines encompass both shader UIDs and render states, changes to either of these should
 // also increment the pipeline UID version. Incrementing the UID version will cause all UID
 // caches to be invalidated.
-constexpr u32 GX_PIPELINE_UID_VERSION = 6;  // Last changed in PR 10890
+constexpr u32 GX_PIPELINE_UID_VERSION = 7;  // Last changed in PR 11859
```
Comment (TellowKrinkle) [inline Source/Core/VideoCommon/GXPipelineTypes.h:22]: Technically, adding a new field to an existing bitfield where zero is the same as the current usage *shouldn't* break any UIDs

**Review by TellowKrinkle — APPROVED** [review]
Up to you whether you want to hope no one somehow ended up with nonzero additional color attachments in their UIDs

**Comment by AdmiralCurtiss** [comment]
So are we bumping the UID or not?

**Comment by iwubcode** [comment]
@AdmiralCurtiss -

FYI I asked Tellow what they meant by their comment and this was their response:

> As in, I can't deny the possibility that we accidentally left garbage in the padding in some code somewhere
So if you're worried about that, then invalidating UIDs would be a good idea

I wouldn't say I'm worried but I didn't think it hurt anything, so figured why not be safe?

**Pull request merged**